src/yunohost/diagnosis.py

```diff
@@ -465,9 +465,9 @@ def diagnose(self):
         new_report = {"id": self.id_, "cached_for": self.cache_duration, "items": items}
 
         self.logger_debug("Updating cache %s" % self.cache_file)
-        self.write_cache(new_report)
         Diagnoser.i18n(new_report)
         add_ignore_flag_to_issues(new_report)
+        self.write_cache(new_report)
 
```

Comment:
Eeeeeh not sure about this change ? The fact that we do the i18n "rendering" with `Diagnoser.i18n(new_report)` *after* we wrote the report is a pretty big deal

Basically we want to save a locale-agnostic diagnosis report because we can't be sure that the locale we'll want to display the report with is the same as the locale used when the report was generated (for example, because the diagnosis is ran from a cron job, etc..)

But I'm assuming that the purpose of this change was related to the `add_ignore_flag_to_issues`  because you want to be able to access the ignore flag in the certificate.py code ... But for the same kind of reason we probably don't want to do that here

I think instead we should either : 
- call add_ignore_flag_to_issues in the relevant certificate.py code
- or maybe rely on diagnosis_show instead of diagnosis_get in the certificate.py code